web/src/scripts/components/editor/Editor.jsx

```diff
@@ -25,28 +25,67 @@ import CodeMirrorComponent from './CodeMirrorComponent'
 class Editor extends Component {
   constructor(props) {
     super(props)
-    this.attachMiddleware(props.middleware, props.decoDoc)
+
+    this.state = {
+      linkedDoc: this.createLinkedDoc(props.decoDoc)
+    }
+
+    this.attachMiddleware(props.middleware, props.decoDoc, this.state.linkedDoc)
+  }
+
+  createLinkedDoc(decoDoc) {
```

Comment:
Hmm.. does it make sense to have this logic in Editor? Where would we put it if not?